pennylane/transforms/qcut.py

```diff
@@ -365,3 +365,132 @@ def graph_to_tape(graph: MultiDiGraph) -> QuantumTape:
                 wire_map[measured_wire] = new_wire
 
     return tape
+
+
+def _prep_zero_state(wire):
+    Identity(wire)
+
+
+def _prep_one_state(wire):
+    PauliX(wire)
+
+
+def _prep_plus_state(wire):
+    Hadamard(wire)
+
+
+def _prep_iplus_state(wire):
+    Hadamard(wire)
+    S(wires=wire)
+
+
+PREPARE_SETTINGS = [_prep_zero_state, _prep_one_state, _prep_plus_state, _prep_iplus_state]
+MEASURE_SETTINGS = [Identity, PauliX, PauliY, PauliZ]
+
+
+def expand_fragment_tapes(
+    tape: QuantumTape,
+) -> Tuple[List[QuantumTape], List[PrepareNode], List[MeasureNode]]:
+    """
+    Expands a fragment tape into a tape for each configuration.
+
+    Args:
+        tape (QuantumTape): the fragment tape to be expanded.
+
+    Returns:
+        Tuple[List[QuantumTape], List[PrepareNode], List[MeasureNode]]: the
+        tapes corresponding to each configration, the preparation nodes and
+        the measurement nodes.
+
+    **Example**
+
+    Consider the following where ``graph`` contains a single
+    :class:`~.MeasureNode` and :class:`~.PrepareNode` within the full circuit
+    graph:
+
+    .. code-block:: python
+
+        from pennylane.transforms import qcut
+
+        >>> subgraphs, communication_graph = qcut.fragment_graph(graph)
+        >>> tapes = [qcut.graph_to_tape(sg) for sg in subgraphs]
+
+        >>> fragment_configurations = [qcut.expand_fragment_tapes(tape) for tape in tapes]
+        >>> fragment_configurations
+        [([<QuantumTape: wires=[0, 1], params=4>,
+           <QuantumTape: wires=[0, 1], params=4>,
+           <QuantumTape: wires=[0, 1], params=4>,
+           <QuantumTape: wires=[0, 1], params=4>],
+          [],
+          [MeasureNode(wires=[1])]),
+         ([<QuantumTape: wires=[1, 2], params=2>,
+           <QuantumTape: wires=[1, 2], params=2>,
+           <QuantumTape: wires=[1, 2], params=2>,
+           <QuantumTape: wires=[1, 2], params=2>],
+          [PrepareNode(wires=[1])],
+          [])]
+
+    """
+    obs_map = {"Identity": Identity, "PauliX": PauliX, "PauliY": PauliY, "PauliZ": PauliZ}
+
+    prepare_nodes = [o for o in tape.operations if isinstance(o, PrepareNode)]
+    measure_nodes = [o for o in tape.operations if isinstance(o, MeasureNode)]
+
+    prepare_combinations = product(range(len(PREPARE_SETTINGS)), repeat=len(prepare_nodes))
+    measure_combinations = product(range(len(MEASURE_SETTINGS)), repeat=len(measure_nodes))
+
+    tapes = []
```

Comment:
Before spending too long on tests - we planned a significant change from the prototype function we have due to using the [grouping](https://pennylane.readthedocs.io/en/stable/code/qml_grouping.html) module to perform 3 measurements per wire instead of 4. This will involve expanding the product of strings `["X", "Y", "Z", "I"]` over the qubits, applying [string_to_pauli_word](https://pennylane.readthedocs.io/en/stable/code/api/pennylane.grouping.string_to_pauli_word.html) to each element, and calling [group_observables](https://pennylane.readthedocs.io/en/stable/code/api/pennylane.grouping.group_observables.html) to give us the final collection of observables we'll need to measure.